### ISSUE TITLE
Shift to using indexed-traversable ?

### ISSUE THREAD

**Issue by cartazio** [bgamari/monoidal-containers#73]
I think it seems that this package is depending on lens for the various flavors of the indexed folds etc.  those will be via indexed-traversable package as of the next lens release I think?  
cc @phadej @emilypi @bgamari

**Comment by emilypi** [comment]
I'd have to defer to Oleg. I don't know if `lens` re-exports the `*WithIndex` classes with the next version. If not, then yes, this package will pick up an additional dependency.

**Comment by phadej** [comment]
I'd recommend to **not** support `lens-4` and `lens-5` (which will use `indexed-traversable`). Or then you'd need to write

```haskell

instance FunctorWithIndex k (MonoidalMap k v) where ....

#if !MIN_VERSION_lens(5,0,0)
instance Lens.FunctorWithIndex k (MonoidalMap k v) where ...
#endif
```

See a PR to `insert-ordered-containers` for an example https://github.com/phadej/insert-ordered-containers/pull/39/files (it uses `lens-4.20` as cutoff, but next lens will be 5).

---

TL;DR `Control.Lens` re-exports `FunctorWithIndex` so you'll get `imap` without extra-dependency, but when you define instance it is advised to depend directly on `indexed-traversable`.

**Comment by endgame** [comment]
The package also provides instances for the `lens` classes `At`, `Empty`, `Ixed` and `Wrapped`, which remain in `lens` AFAIK.
